pipeline/cleanup/cleanup-template.yaml

```diff
@@ -6,40 +6,102 @@ spec:
   params:
   - name: namespace
     description: Namespace to cleanup to in the target cluster
-  - name: clusterResource
-    description: Name of the cluster resource that points to the target cluster
   - name: keep
     description: Amount of old resources to keep
     default: "200"
+  - name: filename
+    description: Filename for the kubeconfig file
+    default: "kubeconfig"
+  - name: name
+    description: Name of the cluster
+  - name: url
+    description: Address of the target cluster
+  - name: username
+    description: Username for basic authentication to the cluster
+  - name: cadata
+    description: Contains PEM-encoded certificate authority certificates
+    default: ""
+  - name: clientKeyData
+    description: Contains PEM-encoded data from a client key file for TLS
+    default: ""
+  - name: clientCertificateData
+    description: Contains PEM-encoded data from a client cert file for TLS
+    default: ""
   resourcetemplates:
   - apiVersion: tekton.dev/v1beta1
-    kind: TaskRun
+    kind: PipelineRun
     metadata:
       name: cleanup-runs-$(tt.params.namespace)-$(uid)
     spec:
       serviceAccountName: tekton-cleaner
-      taskSpec:
-        params:
+      pipelineSpec:
+        workspaces:
+          - name: shared-workspace
+        tasks:
+          - name: kubeconfig-creator
+            taskRef: kubeconfig-creator
```

Comment:
Uhm, it really surprises me that the `kubeconfig-creator` task would take secrets like the token as ca data as parameters.